src/http_server.cc

```diff
@@ -296,29 +336,27 @@ JsonBytesArrayByteSize(
     triton::common::TritonJson::Value& tensor_data, size_t* byte_size)
 {
   *byte_size = 0;
-
-  for (size_t i = 0; i < tensor_data.ArraySize(); i++) {
-    triton::common::TritonJson::Value el;
-    RETURN_IF_ERR(tensor_data.At(i, &el));
-
-    // Recurse if not last dimension...
-    TRITONSERVER_Error* assert_err =
-        el.AssertType(triton::common::TritonJson::ValueType::ARRAY);
-    if (assert_err == nullptr) {
+  // Recurse if not last dimension...
+  TRITONSERVER_Error* assert_err =
```

Comment:
Question / Future Scope:

Not sure why we use the assertion here instead of `IsArray()` with the error we have to clean up , etc. - do we need it?

https://github.com/triton-inference-server/common/blob/cf617c93c75d8e03da962115a2471e4b89062aae/include/triton/common/triton_json.h#L941